goss-testing/scripts/check_bgp_neighbors_established.sh

```diff
@@ -49,19 +49,46 @@ function err_exit
+else
+    kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l >> $TMPFILE ||
+        err_exit 20 "ERROR: Command failed (rc=$?): kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l" > TMPFILE
+	metallb_check=$(kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l)
+fi
+
+# check SLS Networks data for BiCAN toggle
+curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l >> $TMPFILE ||
+    err_exit 30 "ERROR: Command failed (rc=$?): curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l" >> $TMPFILE
+sls_network_check=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l)
+
+# check spine switch motd for csm version
+ssh -o ConnectTimeout=1 -o BatchMode=yes-o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' >> $TMPFILE ||
+    err_exit 40 "ERROR: Command failed (rc=$?): ssh -o ConnectTimeout=1 -o BatchMode=yes-o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' " >> $TMPFILE
+switch_motd_check=$(ssh -o ConnectTimeout=1 -o BatchMode=yes -o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' )
 
 # We really shouldn't be passing a password in plaintext on the command line, but as long as we are, at least
 # we won't also include it in the error message on failure
-if grep -q customer-management $TMPFILE ; then
-    echo "Running: canu validate network bgp --network all --password XXXXXXXX"
-    canu validate network bgp --network all --password {{.Env.SW_ADMIN_PASSWORD}} ||
-        err_exit 20 "canu validate network bgp --network all failed (rc=$?)"
-else
+# RFE: https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-880
+
+if [ "$metallb_check" -eq "0" ] && [ "$sls_network_check" -eq "0" ] || [ "$switch_motd_check" == "1.0" ];then
+    # csm-1.0 networking
     echo "Running: canu validate network bgp --network nmn --password XXXXXXXX"
     canu validate network bgp --network nmn --password {{.Env.SW_ADMIN_PASSWORD}} ||
-        err_exit 30 "canu validate network bgp --network nmn failed (rc=$?)"
+        err_exit 50 "canu validate network bgp --network nmn failed (rc=$?)"
+else
+    # csm-1.2+ networking
+    echo "Running: canu validate network bgp --network all --password XXXXXXXX"
+    canu validate network bgp --network all --password {{.Env.SW_ADMIN_PASSWORD}} ||
+        err_exit 60 "canu validate network bgp --network all failed (rc=$?)"
```

Comment:
With all of the above worked around, I'm still getting a failure logging in to the switches so this CASMINST-4284 is still broken.

```
ncn-m001:~/johren # ./test.sh
Running: canu validate network bgp --network nmn --password XXXXXXXX
Error connecting to switch 10.254.0.2, HTTPError.
/Error connecting to switch 10.254.0.3, HTTPError.


BGP Neighbors Established
--------------------------------------------------


Errors
--------------------------------------------------
10.254.0.2      - Connection Error
10.254.0.3      - Connection Error
PASS
```

Reply:
Added logic to allow manual entry of password if goss env password is null or locally running test outside of goss.